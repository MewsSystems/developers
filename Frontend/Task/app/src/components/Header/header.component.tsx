import React from 'react'
import './styles.module.css'

const Header: React.FC = () => {
  return(
    <header className="header">
      <h1 className="heading">Currency rates exchange</h1>
    </header>
  )
}

export default Header;