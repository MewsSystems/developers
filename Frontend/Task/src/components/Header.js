import React from 'react';
import { Link } from 'react-router-dom';

export const Header = () => (
  <header>
    <Link to="/">
      <h1>Boilerplate</h1>
    </Link>
  </header>
);



export default Header;
