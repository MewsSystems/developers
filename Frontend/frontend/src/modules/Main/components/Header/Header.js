import React from 'react';

import Container from '../../../../atoms/Container/Container';

import StyledHeader from './styles/StyledHeader';


const Header = () => (
  <StyledHeader>
    <Container>
      <div className="header--container">

        <span className="header--title">AppName</span>

      </div>
    </Container>
  </StyledHeader>
);


export default Header;
