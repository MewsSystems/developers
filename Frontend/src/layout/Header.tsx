import React from 'react'
import { NavLink } from 'react-router-dom'
import styled from '../utils/styled'
import Container from './Container'

interface HeaderProps {
  title: string
}

const Wrapper = styled.header`
  padding: 0.5rem 1.5rem;
  background-color: ${props => props.theme.colors.brand};
  color: ${props => props.theme.colors.white};
  font-family: ${props => props.theme.fonts.headings};
`

const HeaderInner = styled(Container)`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;

  @media (min-width: ${props => props.theme.breakpoints.lg}) {
    flex-direction: row;
  }
`

const HeaderLeft = styled.div`
  padding-right: 2rem;
`

const HeaderNav = styled.nav`
  flex: 1 1 auto;
  margin: 1rem 0;

  @media (min-width: ${props => props.theme.breakpoints.lg}) {
    margin: 0;
  }
`

const HeaderNavLink = styled(NavLink)`
  margin: 0 1rem;

  &.is-active {
    text-decoration: underline;
  }
`

const HeaderRight = styled('div')`
  padding-left: 1rem;
`

const Title = styled('h2')`
  margin: 0;
  font-weight: 500;
`

const CurrentTheme = styled('span')`
  margin-right: 1rem;
`

const Header: React.SFC<HeaderProps> = ({ title }) => (
  <Wrapper>
    <HeaderInner>
      <HeaderLeft>
        <Title>{title}</Title>
      </HeaderLeft>
      <HeaderNav>
        <HeaderNavLink to="/movies" activeClassName="is-active">
          Movies
        </HeaderNavLink>
        <HeaderNavLink to="/about" activeClassName="is-active">
          About
        </HeaderNavLink>
      </HeaderNav>
      <HeaderRight>
        <CurrentTheme>for Mews Systems</CurrentTheme>
      </HeaderRight>
    </HeaderInner>
  </Wrapper>
)

export default Header
