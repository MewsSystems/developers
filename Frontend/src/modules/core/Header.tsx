import React from 'react';
import Title from './components/Title';

interface Props {
    className?: string,
}

export default function Header({ className }: Props) {
    return (
        <header className={className}>
            <Title />
        </header>
    );
}
