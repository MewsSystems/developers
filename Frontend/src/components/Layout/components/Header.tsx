import { COLORS } from 'constants/colors'
import { BOX_SHADOW } from 'constants/index'
import { Search } from 'components/Search'
import { Spacing } from 'components/Spacing'
import { StyledInput } from 'components/Input'
import { Navigation, NavigationItemProps } from 'components/Navigation'
import { Logo } from '../../Logo'
import styled from 'styled-components'
import React from 'react'
import { useTranslation } from 'react-i18next'

const Top = styled(Spacing)`
  display: grid;
  grid-gap: 2rem;
  grid-template-columns: min-content auto;
`

const StyledHeader = styled.div`
  padding: 1.6rem 2rem 2rem;
  background: ${COLORS.BLACK};
  box-shadow: ${BOX_SHADOW.MEDIUM};
`

const StyledNavigation = styled(Navigation)`
  align-self: end;
  justify-self: end;

  * {
    font-size: 1.1rem;
  }
`

const StyledSearch = styled(Search)`
  ${StyledInput} {
    color: ${COLORS.WHITE};
    background: rgba(255, 255, 255, 0.3);

    ::placeholder {
      color: ${COLORS.WHITE};
    }
  }
`

const { PUBLIC_URL } = process.env

export const Header: React.FC = () => {
  const { t } = useTranslation()

  const navigationItems: NavigationItemProps[] = [
    {
      title: t('header.navigation.home'),
      to: `${PUBLIC_URL}/`,
      key: 'home',
    },
    {
      title: t('header.navigation.favourite'),
      to: `${PUBLIC_URL}/favourite`,
      key: 'favourite',
    },

    {
      title: t('header.navigation.recomended'),
      to: `${PUBLIC_URL}/recomended`,
      key: 'recomended',
    },
  ]

  return (
    <StyledHeader>
      <Top outer="0 0 1rem">
        <Logo>{t('header.title')}</Logo>
        <StyledNavigation items={navigationItems} />
      </Top>
      <StyledSearch />
    </StyledHeader>
  )
}
