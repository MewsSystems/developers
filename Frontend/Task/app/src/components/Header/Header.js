import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles(theme => ({
    root: {
        flexGrow: 1,
    },
    menuButton: {
        marginRight: theme.spacing(2),
    },
    title: {
        flexGrow: 1,
    },
}));

export default function Header() {
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <AppBar position="static">
                <Toolbar>
                    <Typography variant="h6" className={classes.title}>
                        Currencies
                    </Typography>
                </Toolbar>
            </AppBar>
        </div>
    );
}
