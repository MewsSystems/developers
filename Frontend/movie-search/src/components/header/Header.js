import React from "react";
import { HeaderWrapper } from './Header.styles';
import { withRouter } from 'react-router-dom';

const Header = (props) => {
  return (
    <HeaderWrapper onClick={ () => props.history.push('/') }>
      { props.text }
    </HeaderWrapper>
  );
};

export default withRouter(Header);
