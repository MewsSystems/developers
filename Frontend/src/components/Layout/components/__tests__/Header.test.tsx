import { Header } from '../Header'
import React from 'react'
import { shallow } from 'enzyme'

describe('Test Header', () => {
  it('Tests snapshot', () => {
    expect(shallow(<Header />)).toMatchSnapshot()
  })
})
