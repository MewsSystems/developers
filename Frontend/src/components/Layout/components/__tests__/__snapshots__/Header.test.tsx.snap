// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header Tests snapshot 1`] = `
<styled.div>
  <Styled(Spacing)
    outer="0 0 1rem"
  >
    <styled.span>
      header.title
    </styled.span>
    <Styled(Navigation)
      items={
        Array [
          Object {
            "key": "home",
            "title": "header.navigation.home",
            "to": "/",
          },
          Object {
            "key": "favourite",
            "title": "header.navigation.favourite",
            "to": "/favourite",
          },
          Object {
            "key": "recomended",
            "title": "header.navigation.recomended",
            "to": "/recomended",
          },
        ]
      }
    />
  </Styled(Spacing)>
  <Styled(Search) />
</styled.div>
`;
